### ISSUE TITLE
Set deprecated property for operations that have been deprecated

### ISSUE THREAD

**Issue by darrelmiller** [microsoft/OpenAPI.NET.OData#113]

**Comment by baywet** [comment]
note: here is the annotation we're looking to translate

```xml
<Annotation Term="Org.OData.Core.V1.Revisions">
          <Collection>
            <Record>
              <PropertyValue Date="2021-08-24" Property="Date" />
              <PropertyValue Property="Description" String="The identityProvider API is deprecated and will stop returning data on March 2023. Please use the new identityProviderBase API." />
              <PropertyValue Property="Kind">
                <EnumMember>Org.OData.Core.V1.RevisionKind/Deprecated</EnumMember>
              </PropertyValue>
              <PropertyValue Date="2023-03-15" Property="RemovalDate" />
              <PropertyValue Property="Version" String="2021-05/identityProvider" />
            </Record>
          </Collection>
        </Annotation>
```

It should translate into a `deprecated: true` property on the operation/path item.

Related work https://github.com/microsoft/Vipr/pull/146

**Comment by baywet** [comment]
additional information https://github.com/oasis-tcs/odata-vocabularies/blob/main/vocabularies/Org.OData.Core.V1.md#RevisionType

**Comment by baywet** [comment]
openapi.net currently only supports OpenAPI 3.0 which only supports json schema draft 4, which doesn't have a [deprecated property](https://json-schema.org/draft/2020-12/json-schema-validation.html#rfc.section.9.3) (added in draft 9) on schemas or properties.
OpenAPI 3.1 supports draft 12, and work is in progress to add support for it in OpenAPI.net.

For the time being the conversion process will add the deprecated property and the x- extension only on operations. It will do so by propagating the information to derived types, entity sets, navigation properties and singletons. 

At a later time, when OpenAPI 3.1 is supported by openapi.net, the conversion process might add the deprecated property and x- extension to schemas/properties as well.

**Comment by baywet** [comment]
fixed via #155
